src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -114,33 +112,32 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
                     continue;
                 }
 
-                if (parameterDescription.Source == BindingSource.Body)
-                {
-                    hasJsonBody = true;
-                }
-
                 apiDescription.ParameterDescriptions.Add(parameterDescription);
             }
 
-            // Get custom attributes for the handler. ConsumesAttribute is one of the examples. 
-            var acceptsRequestType = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>()?.RequestType;
-            if (acceptsRequestType is not null)
+            // Get IAcceptsMetadata.
+            var acceptsMetadata = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>();
+            if (acceptsMetadata is not null)
             {
-                var parameterDescription = new ApiParameterDescription
+                var acceptsRequestType = acceptsMetadata.RequestType;
+                var isOptional = acceptsMetadata.IsOptional;
+                if (acceptsRequestType is not null)
```

Comment:
I think we should just create the best `ApiParameterDescription` we can based on the metadata even if it's missing the parameter type. `ApiParameterDescription.Type` is non-nullable but we can just do `Type = acceptsRequestType ?? typeof(void)` below.

Reply:
Do we really want this behavior below ?  Is the typeof(Void) meaningful to the user ? Also what do you send in the Description Name ? Null ? (Swashbuckle will throw on Null)

![image](https://user-images.githubusercontent.com/87031580/131932894-7f23214f-30dc-449d-ab08-ef6406d23a68.png)

Reply:
I think that behavior is *much* better than just ignoring that "applicaiton/xml" was specified at all. It makes it a lot clearer to the user that there metadata was successfully added to the endpoint, but they didn't provide any information about the type.

I'd make the Name "Void". That's what `typeof(void).Name` would evaluate to anyway, so it's consistent.